packages/common/src/lib/common/tools.ts

```diff
@@ -3084,6 +3084,9 @@ export function maybeCallback(
     }
 }
 
+type MaybeCbErrCallback = ((error: Error | null | undefined, ...args: any[]) => void) | null | undefined;
```

Comment:
We could make this smarter, so `...args` gets inferred from the actual callback's argument types.
I can take care of this if we want this.

Reply:
If you guide me in the right direction, maybe I will be able to do it too and of course learn something ;-)